product/views.py

```diff
@@ -4,9 +4,36 @@
 from django.http  import JsonResponse
 from django.db    import transaction, IntegrityError
 
-from .models      import MatchingReview, Review
-from order.models import Order
-from utils         import login_decorator
+from user.models    import User
+from order.models   import Order
+from product.models import ProductLike, Product, MatchingReview, Review
+from utils          import login_decorator
+
+class ProductLikeView(View):
+    @login_decorator
+    def get(self, request):
+        
+        likes = ProductLike.objects.filter(user_id = request.user.id)
+        likes = [{
+            'id'       : like.product.id,
+            'name'     : like.product.name,
+            'imageUrl' : like.product.image_url,
+            'category' : like.product.category.id
+        } for like in likes]g
+
+        return JsonResponse({'message':'SUCCESS', 'data':likes}, status=200)
+    
+    @login_decorator
+    def post(self, request):
+        data = json.loads(request.body)
+
+        check_have  = ProductLike.objects.get_or_create(user_id=request.user.id, product_id=data['productId'])
+        have_like   = check_have[0]
+        is_have_like = check_have[1]
+        
```

Comment:
```suggestion

        have_like, is_have_like  = ProductLike.objects.get_or_create(user_id=request.user.id, product_id=data['productId'])
        
```
과 같은 방법으로도 사용하실 수 있습니다 :)

Reply:
앗 감사합니다! 바로 적용했습니다!!